### ISSUE TITLE
Is data augmentation applied to validation set?

### ISSUE THREAD

**Issue by HannahAlexander** [ultralytics/ultralytics#9874]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I just wanted to check that the in built data augmentation available in yolov8 is only applied to the train set and not the validation set.
thanks

**Comment by glenn-jocher** [comment]
@HannahAlexander hi there! 😊

That's correct, the built-in data augmentation in YOLOv8 is applied exclusively to the training set. The validation set is not augmented and is used as-is to evaluate the model's performance accurately.

For instance, when training:

```python
model.train(data='train_data.yaml', imgsz=640, epochs=100)
```

Data augmentation is applied to the training dataset specified in `train_data.yaml`. During validation:

```python
model.val(data='val_data.yaml')
```

The validation dataset (`val_data.yaml`) is used without any augmentation.

Hope this clarifies your query! If you have more questions, feel free to ask.

**Comment by August9708** [comment]
> @HannahAlexander hi there! 😊
> 
> That's correct, the built-in data augmentation in YOLOv8 is applied exclusively to the training set. The validation set is not augmented and is used as-is to evaluate the model's performance accurately.
> 
> For instance, when training:
> 
> ```python
> model.train(data='train_data.yaml', imgsz=640, epochs=100)
> ```
> 
> Data augmentation is applied to the training dataset specified in `train_data.yaml`. During validation:
> 
> ```python
> model.val(data='val_data.yaml')
> ```
> 
> The validation dataset (`val_data.yaml`) is used without any augmentation.
> 
> Hope this clarifies your query! If you have more questions, feel free to ask.

If I hope that train set is also not augmented, what should I do?

**Comment by August9708** [comment]
> @HannahAlexander hi there! 😊
> 
> That's correct, the built-in data augmentation in YOLOv8 is applied exclusively to the training set. The validation set is not augmented and is used as-is to evaluate the model's performance accurately.
> 
> For instance, when training:
> 
> ```python
> model.train(data='train_data.yaml', imgsz=640, epochs=100)
> ```
> 
> Data augmentation is applied to the training dataset specified in `train_data.yaml`. During validation:
> 
> ```python
> model.val(data='val_data.yaml')
> ```
> 
> The validation dataset (`val_data.yaml`) is used without any augmentation.
> 
> Hope this clarifies your query! If you have more questions, feel free to ask.

![image](https://github.com/ultralytics/ultralytics/assets/78811371/b81e23a0-f5a7-4fb0-b15f-6512d6636287)
If I hope that The train dataset is also used without any augmentation, what should I do?

**Comment by glenn-jocher** [comment]
Hey there! 😊

To use your training dataset without any augmentation in YOLOv8, you can adjust your dataset configuration file (e.g., `train_data.yaml`). Within that file, ensure no augmentation parameters are enabled or set their values so that they effectively don't alter your data. By default, if augmentation parameters are not specified, the data should not undergo augmentation.

Here's a quick example of how your `train_data.yaml` might look to avoid augmentations:

```yaml
train: path/to/train/images
val: path/to/val/images

# No augmentation parameters specified, so training data used as-is.
```

However, keep in mind that using data augmentation is generally beneficial for improving model performance by increasing dataset diversity. If you have specific reasons for avoiding augmentation, you might want to conduct experiments to compare the model's performance with and without augmentations.

Hope this helps! If you have any other questions, just shoot!
